sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -152,7 +154,7 @@ public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse
      */
     public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation,
                   Supplier<Mono<T>> activationOperation, Consumer<Poller<T>> cancelOperation) {
-        if (pollInterval == null || pollInterval.toNanos() <= 0) {
+        if (pollInterval == null || pollInterval.compareTo(Duration.ZERO) <= 0) {
```

Comment:
If it's null, then we must throw NPE and if the value is invalid, throw `IllegalArgumentException`.

Reply:
I think underneath ClientLogger API  does this check so it will take care of this `Objects.requireNonNull`